src/CoreLib/DSetGenerics.fs

```diff
@@ -626,6 +628,26 @@ type DSet<'U> () =
         x
 
     /// <summary>
+    /// Create a distributed dataset on the distributed cluster, with each element created by a functional delegate, using
+    /// a given number of parallel execution per node.
+    /// <param name="initFunc"> The functional delegate that create each element in the dataset, the integer index passed to the function 
+    /// indicates the partition, and the second integer passed to the function index (from 0) element within the partition. 
+    /// </param> 
+    /// <param name="partitionSizeFunc"> The functional delegate that returns the size of the partition,  the integer index passed to the function 
+    /// indicates the partition. 
+    /// </param> 
+    /// </summary> 
+    member x.InitN(initFunc, partitionSizeFunc : int->int->int) =
+        x.NumPartitions <- x.Cluster.NumNodes * x.NumParallelExecution
+        x.NumReplications <- 1
+        x.Dependency <- Source
+        x.Function <- Function.Init<'U>( initFunc, partitionSizeFunc x.NumPartitions ) 
+        // Trigger of setting the serialization limit parameter within functions. 
+        // Automatic set serialization limit
+        // x.SerializationLimit <- x.SerializationLimit
+        x
+
+    /// <summary>
```

Comment:
For introducing a new DSet API here, there are a few more places to change
1. Add a corresponding static member in this file
2. Add the corresponding C# API in DSetCSharp.fs